camelot/admin/entity_admin.py

```diff
@@ -644,6 +645,54 @@ def get_search_fields(self, substring):
                     self._search_fields.append(field_name)
         return self._search_fields
 
+    def decorate_search_query(self, query, text):
+        """
+        Decorate the given sqlalchemy query for the objects that should be displayed in the table or selection view,
+        with the needed clauses for filtering based on the given search text.
+        By default all 'simple' columns of this admin's and the explicitly set search fields will be used to compare the search text with.
+        Overwrite this method to change this behaviour with more fine-grained or complex search strategies.
+        """
+        if (text is not None) and len(text.strip()):
+            # arguments for the where clause
+            args = []
+            # join conditions : list of join entities
+            joins = []
+    
+            for t in text.split(' '):
+                subexp = []
+                for column_name in self._get_search_fields(t):
+                    path = column_name.split('.')
+                    target = self.entity
+                    related_admin = self
+                    for path_segment in path:
+                        # use the field attributes for the introspection, as these
+                        # have detected hybrid properties
+                        fa = related_admin.get_descriptor_field_attributes(path_segment)
+                        instrumented_attribute = getattr(target, path_segment)
+                        if fa.get('target', False):
+                            joins.append(instrumented_attribute)
+                            target = fa['target']
+                            related_admin = related_admin.get_related_admin(target)
+                        else:
+                            # Append a search clause for the column using a set search strategy, or the basic strategy by default.
+                            fa = related_admin.get_field_attributes(instrumented_attribute.key)
+                            search_strategy = fa.get('search_strategy', related_admin.search_strategy)
```

Comment:
ik zou de get_field_attributes aanpassen, zodat die search_strategy altijd gezet is als field attribute, en er dan hier van uit gaan dat die er is, dat is conform aan hoe de andere fa werken

Reply:
Ok, zie  #56.